src/mlpack/core/cv/metrics/r2_score_impl.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file r2_score_impl.hpp
+ * @author Bisakh Mondal
+ *
+ * The implementation of the class R2Score.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_R2SCORE_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_R2SCORE_IMPL_HPP
+
+namespace mlpack {
+namespace cv {
+
+template<typename MLAlgorithm, typename DataType, typename ResponsesType>
+double R2Score::Evaluate(MLAlgorithm& model,
+                         const DataType& data,
+                         const ResponsesType& responses)
+{
+  if (data.n_cols != responses.n_cols)
+  {
+    std::ostringstream oss;
+    oss << "R2Score::Evaluate(): number of points (" << data.n_cols << ") "
```

Comment:
This is something I am not sure about. @favre49, Let me know what you think. Rather than using ostringstream, why don't we Log::Warn or Log::Assert, maybe Log::Fatal.

Reply:
I had the same doubt, meant to ask on the IRC about it but it slipped my mind. While this is consistent with the the metrics, it isn't with the rest of the codebase.

Reply:
Also, thanks for leaving this review, I often miss out on some things :)

Reply:
No worries. Thanks a lot for all the help on my and the rest of the PRs. :)
Hey @zoq, Could you help us out with this comment.

Reply:
I also think using Log will be a good choice. In this PR I would love to keep it maintaining the design with other metrics. Would love to change it across all metrics in another PR. 
Thanks a lot guys for the wonderful reviews.